runtime/fieldmask.go

```diff
@@ -0,0 +1,66 @@
+package runtime
+
+import (
+	"io"
+	"encoding/json"
+	"google.golang.org/genproto/protobuf/field_mask"
+	"strings"
+	"github.com/golang/protobuf/protoc-gen-go/generator"
+)
+
+// FieldMaskFromRequestBody creates a FieldMask printing all complete paths from the JSON body.
+func FieldMaskFromRequestBody(r io.Reader) (*field_mask.FieldMask, error) {
+	fm := &field_mask.FieldMask{}
+	var root interface{}
+	if err := json.NewDecoder(r).Decode(&root); err != nil {
+		if err == io.EOF {
+			return fm, nil
+		}
+		return nil, err
+	}
+
+	queue := []struct {
```

Comment:
Can you extract the struct into a named type (since you use it below) but leave it unexported